src/FtpExtractor.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\FtpExtractor;
+
+use League\Flysystem\Filesystem as FtpFilesystem;
+use Webmozart\Glob\Glob;
+
+class FtpExtractor
+{
+    private const RECURSIVE_COPY = true;
+    private const IS_FILE = 'file';
+
+    /**
+     * @var FtpFilesystem
+     */
+    private $ftpFilesystem;
+
+    /**
+     * @var bool
+     */
+    private $onlyNewFiles;
+
+    /**
+     * @var bool
+     */
+    private $isWildcard;
+
+    public function __construct(Config $config, FtpFilesystem $ftpFs)
+    {
+        $this->ftpFilesystem = $ftpFs;
+        $this->onlyNewFiles = $config->isOnlyForNewFiles();
+        $this->isWildcard = $config->isWildcard();
+    }
+
+    public function copyFiles(string $sourcePath, string $destionationPath, FileStateRegistry $registry): int
+    {
+        if ($this->isWildcard) {
+            $cnt = $this->downloadFolder($sourcePath, $destionationPath, $registry);
+        } else {
+            $cnt = $this->downloadSingleFile($sourcePath, $destionationPath, $registry);
+        }
+
+        $registry->saveState();
```

Comment:
tady je vidět, že ti dovnitř jinak úplně oddělené komponenty leakuje něco z Connection (něco co ví, že se connection nějak chová a někam něco zapisuje)

Reply:
tady je to místo, kde mi to smrdělo a věděl jsem, že je to trochu špatně:) už to tam není.